conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureBodySerDe.java

```diff
@@ -135,17 +136,45 @@ public void serialize(T value, HttpServerExchange exchange) throws IOException {
     private static final class EncodingDeserializerRegistry<T> implements Deserializer<T> {
 
         private final List<EncodingDeserializerContainer<T>> encodings;
+        private final boolean optionalType;
+        private final TypeMarker<T> marker;
```

Comment:
Alternatively we could store `@Nullable private final T optionalType` instead of these two fields. Doesn't make much difference either way.